File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(!head){
            return nullptr;
        }
        ListNode* curr= head;
        vector<int> arr;
        while(curr){
            arr.push_back(curr->val);
            curr= curr->next;
        }
        reverse(arr.begin(), arr.end());
        ListNode* newNode= new ListNode(arr[0]);
        ListNode* track= newNode;
        for(int i= 1; i<arr.size(); i++){
            track->next= new ListNode(arr[i]);
            track= track->next;
        }
        return newNode;
    }
};

File: Data Structures & Algorithms/reorder-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public void reorderList(ListNode head) {
        ArrayList<Integer> arr= new ArrayList<>();
        ListNode curr= head;
        while(curr!= null){
            arr.add(curr.val);
            curr= curr.next;
        }
        int i= 0;
        int j= arr.size()-1;
        ArrayList<Integer> arr2= new ArrayList<>();
        int k= 0;
        while(i<=j){
            arr2.add(arr.get(i));
            arr2.add(arr.get(j));
            i++;
            j--;
        }
        ListNode ans= head;
        while(ans!= null){
            ans.val= arr2.get(k);
            k++;
            ans= ans.next;
        }

    }
}
